massa-execution/src/interface_impl.rs

```diff
@@ -145,41 +166,92 @@ impl Interface for InterfaceImpl {
         };
-        let key = massa_hash::hash::Hash::from_bs58_check(key)?;
+        let key = massa_hash::hash::Hash::compute_from(key.as_bytes());
         match context.ledger_step.get_data_entry(addr, &key) {
             Some(bytecode) => Ok(bytecode),
             _ => bail!("Data entry not found"),
         }
     }
 
-    fn set_data(&self, key: &str, value: &Bytecode) -> Result<()> {
+    fn set_data(&self, key: &String, value: &Vec<u8>) -> Result<()> {
         let mut context = context_guard!(self);
         let addr = match context.call_stack.back() {
             Some(addr) => *addr,
             _ => bail!("Failed to read call stack current address"),
         };
-        let key = massa_hash::hash::Hash::from_bs58_check(key)?;
+        let key = massa_hash::hash::Hash::compute_from(key.as_bytes());
         context.ledger_step.set_data_entry(addr, key, value.clone());
         Ok(())
     }
 
+    fn has_data(&self, key: &String) -> Result<bool> {
+        let context = context_guard!(self);
+        let addr = match context.call_stack.back() {
+            Some(addr) => addr,
+            _ => bail!("Failed to read call stack current address"),
+        };
+        let key = massa_hash::hash::Hash::compute_from(key.as_bytes());
+        Ok(context.ledger_step.has_data_entry(addr, &key))
+    }
+
+    /// hash data
+    fn hash(&self, data: &Vec<u8>) -> Result<assembly_simulator::MassaHash> {
+        Ok(massa_hash::hash::Hash::compute_from(data).to_bs58_check())
+    }
+
+    /// convert a pubkey to an address
+    fn address_from_public_key(
+        &self,
+        public_key: &assembly_simulator::PublicKey,
+    ) -> Result<assembly_simulator::Address> {
+        let public_key = massa_signature::PublicKey::from_bs58_check(public_key)?;
+        let addr = massa_models::Address::from_public_key(&public_key);
+        Ok(addr.to_bs58_check())
+    }
+
+    /// Verify signature
+    fn signature_verify(
+        &self,
+        data: &Vec<u8>,
+        signature: &assembly_simulator::Signature,
+        public_key: &assembly_simulator::PublicKey,
+    ) -> Result<bool> {
+        let signature = match massa_signature::Signature::from_bs58_check(signature) {
+            Ok(sig) => sig,
+            Err(_) => return Ok(false),
+        };
+        let public_key = match massa_signature::PublicKey::from_bs58_check(public_key) {
+            Ok(pubk) => pubk,
+            Err(_) => return Ok(false),
+        };
+        let h = massa_hash::hash::Hash::compute_from(data);
+        Ok(massa_signature::verify_signature(&h, &signature, &public_key).is_ok())
+    }
+
     /// Transfer coins from the current address to a target address
     /// to_address: target address
     /// raw_amount: amount to transfer (in raw u64)
     fn transfer_coins(&self, to_address: &String, raw_amount: u64) -> Result<()> {
-        let to_address = Address::from_str(to_address)?;
+        let to_address = massa_models::Address::from_str(to_address)?;
```

Comment:
Can we perhaps just import stuff at the top of the file?

Reply:
I did that to be super explicit as there is an ambiguity between massa_models::Address and runtime_simulator::Address, same for PublicKey, Signature etc...